### PR TITLE
Finalize DB Client

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Fixed poms after rebasing to master
JDBC conncetions use HikariCP now.
Atted tests for
- transactions
- mapping
- health check
- metrics
- interceptor
Issue #581

### CODE REVIEW

File: dbclient/common/src/main/java/io/helidon/dbclient/common/AbstractStatement.java [hunk 1]
```diff
@@ -37,6 +38,10 @@
  * @param <R> the result type of the statement as returned by {@link #execute()}
  */
 public abstract class AbstractStatement<S extends DbStatement<S, R>, R> implements DbStatement<S, R> {
+
+    /** Local logger instance. */
+    private static final Logger LOG = Logger.getLogger(AbstractStatement.class.getName());
```
Comment (tomas-langer) [inline dbclient/common/src/main/java/io/helidon/dbclient/common/AbstractStatement.java]: 1. Use the name `LOGGER` to be consistent
2. The instance is never used, so why is it added?
3. Only changes in formatting in this class

File: dbclient/jdbc/pom.xml [hunk 2]
```diff
@@ -49,6 +49,11 @@
             <artifactId>helidon-common-configurable</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.zaxxer</groupId>
+            <artifactId>HikariCP</artifactId>
+            <scope>compile</scope>
```
Comment (tomas-langer) [inline dbclient/jdbc/pom.xml]: Do not use explicit `compile` scope

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcDbClient.java [hunk 3]
```diff
@@ -41,6 +42,10 @@
  * Helidon DB implementation for JDBC drivers.
  */
 class JdbcDbClient implements DbClient {
+
+    /** Local logger instance. */
+    private static final Logger LOG = Logger.getLogger(DbClient.class.getName());
```
Comment (tomas-langer) [inline dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcDbClient.java]: Rename to `LOGGER`

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java [hunk 4]
```diff
@@ -52,7 +52,8 @@
  * Implementation of query.
  */
 class JdbcStatementQuery extends JdbcStatement<DbStatementQuery, DbRows<DbRow>> implements DbStatementQuery {
-    private static final Logger LOGGER = Logger.getLogger(JdbcStatementQuery.class.getName());
+
+    private static final Logger LOG = Logger.getLogger(JdbcStatementQuery.class.getName());
```
Comment (tomas-langer) [inline dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java]: `LOGGER`

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java [hunk 5]
```diff
@@ -66,7 +67,7 @@
 
         executeContext().addFuture(queryFuture);
 
-        return dbContextFuture.thenCompose(interceptorContext -> {
+        CompletionStage<DbRows<DbRow>> result = dbContextFuture.thenCompose(interceptorContext -> {
```
Comment (tomas-langer) [inline dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java]: If you rebase on latest master, you can use Java 11 and `var`, which may be nicer when we use these complex generics (not required)

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java [hunk 6]
```diff
@@ -79,13 +80,19 @@
                                             queryFuture,
                                             rs);
                 } catch (SQLException e) {
+                    LOG.warning(() -> String.format("Query execution failed: %s", e.getMessage()));
```
Comment (tomas-langer) [inline dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java]: Logging a warning AND throwing an exception is not a good combination.
I would log in `FINE` level, including the actual exception (to see the stack trace).
Let the consumer of this future decide what to do with the exception.
Also never log twice in a single block with no operation in between the logs. That should always be a single log statement.

File: dbclient/mongodb/src/main/java/io/helidon/dbclient/mongodb/MongoDbClientProviderBuilder.java [hunk 7]
```diff
@@ -66,9 +67,13 @@ public DbClient build() {
...
     public MongoDbClientProviderBuilder config(Config config) {
-        config.get("url").asString().ifPresent(this::url);
-        config.get("username").asString().ifPresent(this::username);
-        config.get("password").asString().ifPresent(this::password);
+        config.get("connection").asNode().ifPresentOrElse(conn -> {
+            conn.get("url").asString().ifPresent(this::url);
+            conn.get("username").asString().ifPresent(this::username);
+            conn.get("password").asString().ifPresent(this::password);
+        }, () -> {
+            throw new DbClientException("No database connection configuration was found");
```
Comment (tomas-langer) [inline dbclient/mongodb/src/main/java/io/helidon/dbclient/mongodb/MongoDbClientProviderBuilder.java]: You should mention the exact key that is expected. Use `config.get("connection").key()`

**Comment by Tomas-Kraus** [comment]
Most of those things (logging) are on my TODO list anyway. I'll do it before final review.

File: dbclient/common/src/main/java/io/helidon/dbclient/common/AbstractStatement.java (hunk 1 above)
Comment (Tomas-Kraus) [reply dbclient/common/src/main/java/io/helidon/dbclient/common/AbstractStatement.java]: Fixed.

File: dbclient/jdbc/pom.xml (hunk 2 above)
Comment (Tomas-Kraus) [reply dbclient/jdbc/pom.xml]: Fixed.

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcDbClient.java (hunk 3 above)
Comment (Tomas-Kraus) [reply dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcDbClient.java]: Fixed.

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java (hunk 5 above)
Comment (Tomas-Kraus) [reply dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java]: Fixed.

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java (hunk 4 above)
Comment (Tomas-Kraus) [reply dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java]: Fixed.

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java (hunk 5 above)
Comment (Tomas-Kraus) [reply dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java]: I would like to keep full types here now. It's questionable whether code will be better readable with var or full type information.

File: dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java (hunk 6 above)
Comment (Tomas-Kraus) [reply dbclient/jdbc/src/main/java/io/helidon/dbclient/jdbc/JdbcStatementQuery.java]: Changed to finest level.

File: dbclient/mongodb/src/main/java/io/helidon/dbclient/mongodb/MongoDbClientProviderBuilder.java (hunk 7 above)
Comment (Tomas-Kraus) [reply dbclient/mongodb/src/main/java/io/helidon/dbclient/mongodb/MongoDbClientProviderBuilder.java]: Done.

**Pull request merged**